sqlglot/dialects/tsql.py

```diff
@@ -708,6 +715,33 @@ def parse_range():
 
             return partition
 
+        def _parse_declare(self) -> exp.Declare | exp.Command:
+            expressions = self._try_parse(partial(self._parse_csv, self._parse_declareitem))
+            if not expressions:
+                return self._parse_as_command(self._prev)
+            return self.expression(exp.Declare, expressions=expressions)
+
+        def _parse_declareitem(self) -> t.Optional[exp.DeclareItem]:
+            var = self._parse_id_var(any_token=False, tokens=[TokenType.PARAMETER])
```

Comment:
I believe we can simply `_parse_id_var()` here right? No need to constrain this, `_parse_identifier()` will parse the `Parameter` because it calls `_parse_placeholder()` as a fallback.
```suggestion
            var = self._parse_id_var()
```